src/Stack/Build/Installed.hs

```diff
@@ -126,13 +133,15 @@ loadDatabase :: (M env m, PackageInstallInfo pii)
              -> GetInstalledOpts
              -> Maybe InstalledCache -- ^ if Just, profiling or haddock is required
              -> Map PackageName pii -- ^ to determine which installed things we should include
-             -> Maybe (InstallLocation, Path Abs Dir) -- ^ package database, Nothing for global
+             -> [Path Abs Dir] -- ^ extra package databases this database depends on
+             -> Maybe (InstalledPackageLocation, Path Abs Dir) -- ^ package database, Nothing for global
              -> [LoadHelper] -- ^ from parent databases
              -> m ([LoadHelper], [DumpPackage () ()])
-loadDatabase menv opts mcache sourceMap mdb lhs0 = do
+loadDatabase menv opts mcache sourceMap extraDBs mdb lhs0 = do
     wc <- getWhichCompiler
-    (lhs1, dps) <- ghcPkgDump menv wc (fmap snd mdb)
-                  $ conduitDumpPackage =$ sink
+    (lhs1, dps) <- ghcPkgDump menv wc (extraDBs ++ (fmap snd (maybeToList mdb)))
```

Comment:
If you don't pass the extra databases to `ghc-pkg` it will report the packages depending on any of them in the actual database as broken